File: design_patterns/c++/src/create/builder/CountingMazeBuilder.cpp
//
// Created by Albert on 2020/12/10.
//

#include "../../../headers/create/builder/CountingMazeBuilder.h"

CountingMazeBuilder::CountingMazeBuilder() {
  _rooms = _doors = 0;
}

void CountingMazeBuilder::buildRoom(int) {
  _rooms++;
}
void CountingMazeBuilder::buildDoor(int, int) {
  _doors++;
}
void CountingMazeBuilder::getCounts(int &rooms, int &doors) const {
  rooms = _rooms;
  doors = _doors;
}

void CountingMazeBuilder::buildMaze() {

}
void CountingMazeBuilder::addWall(int, Direction) {

}

File: design_patterns/java/src/create/builder/CountingMazeBuilder.java
package create.builder;//
// Created by Albert on 2020/12/10.
//

import base.Maze;
import javafx.util.Pair;

class CountingMazeBuilder extends MazeBuilder {
    private int doors;
    private int rooms;

    @Override
    public Maze buildMaze() {
        return null;
    }

    @Override
    public void buildRoom(int room) {
        rooms++;
    }

    @Override
    public void buildDoor(int roomFrom, int roomTo) {
        doors += 2;
    }

    public Pair<Integer, Integer> getCounts() {
        return new Pair<Integer, Integer>(doors, rooms);
    }

};